src/validation.js

```diff
@@ -17,6 +17,29 @@ dayjs.extend(isSameOrAfter);
 
 export let validatorFns = {};
 
+export const IS_PERCENTAGE = 'validator/IS_PERCENTAGE';
+export const IS_PERCENTAGE_ERROR = 'error/IS_PERCENTAGE';
+export const isPercentage = createValidator(IS_PERCENTAGE, IS_PERCENTAGE_ERROR);
+validatorFns[IS_PERCENTAGE] = (value, args, form) => {
+  if (value === '') {
+    return true;
+  }
+  const parts = value.split('.');
+  if (parts.length > 2) {
+    // Ensure there are 1 or 2 parts (integer and optional decimal)
+    return false;
+  }
+  const decimalPart = parts[1];
+  // If there's a decimal part, ensure it's 1 or 2 digits
+  if (decimalPart) {
+    if (decimalPart.length > 2 || isNaN(decimalPart)) {
+      return false;
+    }
+  }
+  const num = parseFloat(value);
+  return num >= 0 && num <= 100;
+};
```

Comment:
My initial AI prompt for this generic function inspiration was something like `Write a javascript function which calculates whether a stringified number input is a percentage. It can optionally contain a decimal and there can be 1 or 2 digits after the decimal`.

Unfortunately I recently cleared my history to reduce clutter but my first iteration of this function was an altered version of the reply: 

```
const isPercentage = (value) => {
  if (value === '') {
    return true;
  }
  const parts = value.split('.');
  if (parts.length > 2) {
    // Ensure there are 1 or 2 parts (integer and optional decimal)
    return false;
  }
  const decimalPart = parts[1];
  // If there's a decimal part, ensure it's 1 or 2 digits
  if (decimalPart) {
    if (decimalPart.length > 2 || isNaN(decimalPart)) {
      return false;
    }
  }
  const convertedNumber = Number.parseFloat(value).toFixed(2);
  return convertedNumber >= 0 && convertedNumber <= 100;
};
```

Since this contains a lot of logic, I then asked if it could be simplified, which it proposed:

![image](https://github.com/CityBaseInc/redux-freeform/assets/100228853/0273ff28-9f6c-4116-bab8-68eb6f1cdc91)

I ended up using the second suggestion because I agreed it was helpful but not the first because I felt what I had originally reads better to a human.